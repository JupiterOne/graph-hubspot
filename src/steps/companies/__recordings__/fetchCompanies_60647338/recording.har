{
  "log": {
    "_recordingName": "fetchCompanies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "178c54fa7da49daabd661efd712163a1",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            }
          ],
          "headersSize": 387,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "100"
            }
          ],
          "url": "https://api.hubapi.com/crm/v3/objects/companies?limit=100"
        },
        "response": {
          "bodySize": 589,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 589,
            "text": "{\"results\":[{\"id\":\"6534491450\",\"properties\":{\"createdate\":\"2021-07-08T05:09:01.918Z\",\"domain\":\"hubspot.com\",\"hs_lastmodifieddate\":\"2021-07-08T05:10:03.397Z\",\"hs_object_id\":\"6534491450\",\"name\":\"HubSpot, Inc.\"},\"createdAt\":\"2021-07-08T05:09:01.918Z\",\"updatedAt\":\"2021-07-08T05:10:03.397Z\",\"archived\":false},{\"id\":\"6558739580\",\"properties\":{\"createdate\":\"2021-07-13T05:34:55.981Z\",\"domain\":null,\"hs_lastmodifieddate\":\"2021-07-13T05:35:58.142Z\",\"hs_object_id\":\"6558739580\",\"name\":\"Test company\"},\"createdAt\":\"2021-07-13T05:34:55.981Z\",\"updatedAt\":\"2021-07-13T05:35:58.142Z\",\"archived\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 23 Jul 2021 12:41:59 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "6734fc71af87404c-MNL"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "8beafaf1-7d3f-403c-aadb-d7e69c700104"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "97"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "x-trace",
              "value": "2B84998D59240072AA3AF04E74DFF4D086E0CA1DA8000000000000000000"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=QrW8vZPN1Q%2FqopPPvNMzPtN%2FbgHKN85U3ftz2d8Hr98ZiCa%2FnREjUSZmKiidxobFBgMUrKqXeb2BIdrqCcVbGGNxAvxUey2nRCnnt7sprIdP5gguvbUay8qCbXLXmcGd\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1172,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-23T12:42:11.685Z",
        "time": 465,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 465
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
