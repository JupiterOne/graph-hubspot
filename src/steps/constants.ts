import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export enum IntegrationSteps {
  OWNERS = 'fetch-owners',
  COMPANIES = 'fetch-companies',
  DOMAINS = 'fetch-domains',
  TEAMS = 'fetch-teams',
  CONTACTS = 'fetch-contacts',
}

export const Entities: Record<
  'USER' | 'TEAM' | 'CONTACT' | 'COMPANY' | 'DOMAIN',
  StepEntityMetadata
> = {
  /*
     Depending on the other resources later (e.g. will we encounter regular users?)
     We may want to call this "Owner" or similar, we can still use the same class of "User"
     Since user can be both a 'regular' user and the owner.
     A bit hard to tell for sure right now
  */
  USER: {
    _type: 'hubspot_user',
    _class: 'User',
    resourceName: 'HubSpot User',
  },

  TEAM: {
    _type: 'hubspot_team',
    _class: 'Team',
    resourceName: 'HubSpot Team',
  },

  CONTACT: {
    _type: 'hubspot_contact',
    _class: 'User',
    resourceName: 'Hubspot Contact',
  },

  COMPANY: {
    _type: 'hubspot_company',
    _class: 'Organization',
    resourceName: 'HubSpot Company',
  },

  DOMAIN: {
    _type: 'hubspot_domain',
    _class: 'Domain',
    resourceName: 'Hubspot domain',
  },
};

export const Relationships: Record<
  'USER_HAS_TEAM',
  StepRelationshipMetadata
> = {
  USER_HAS_TEAM: {
    _type: 'hubspot_user_has_team',
    _class: RelationshipClass.HAS,
    sourceType: Entities.USER._type,
    targetType: Entities.TEAM._type,
  },
};
