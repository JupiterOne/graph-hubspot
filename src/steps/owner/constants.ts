export const OwnerEntity = {
  RESOURCE_NAME: 'Owner',
  TYPE: 'acme_user',
  CLASS: 'User',
};
