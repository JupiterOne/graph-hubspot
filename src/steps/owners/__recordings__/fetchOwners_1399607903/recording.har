{
  "log": {
    "_recordingName": "fetchOwners",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "3befe0ca00fe7532a452821224793de3",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            },
            {
              "name": "accept",
              "value": "application/json"
            }
          ],
          "headersSize": 376,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/settings/v3/users/roles"
        },
        "response": {
          "bodySize": 202,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 202,
            "text": "{\"results\":[{\"id\":\"95124\",\"name\":\"TestRole\",\"requiresBillingWrite\":false},{\"id\":\"95389\",\"name\":\"SuperRole\",\"requiresBillingWrite\":true},{\"id\":\"96735\",\"name\":\"LimitedRole\",\"requiresBillingWrite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 27 Jul 2021 14:09:57 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "202"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "675672cc2db63cee-HKG"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "d4592025-10b2-4b78-b13d-f7cc3e01a72d"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "99"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=govQi2xUoo5xdrD8uZWOHYHn%2FGGfBqa1EhHsmhgUzWVMkt4khx8amrS8t66VLvc1sBEPt13Ko524fm674wjpzt7R94C1%2FQUL7Wj%2FtZ3Th0mGQlHn1ffx16fp2IrmHu36\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1094,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-27T14:10:05.380Z",
        "time": 543,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 543
        }
      },
      {
        "_id": "986da158beec2c9b278f24e44d023260",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            },
            {
              "name": "accept",
              "value": "application/json"
            }
          ],
          "headersSize": 371,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/settings/v3/users/"
        },
        "response": {
          "bodySize": 175,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 175,
            "text": "{\"results\":[{\"id\":\"25387116\",\"email\":\"pgilaga@gmail.com\",\"primaryTeamId\":\"3785345\"},{\"id\":\"25444566\",\"email\":\"pgrilaga@gmail.com\",\"roleId\":\"96735\",\"primaryTeamId\":\"3785345\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 27 Jul 2021 14:09:58 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "175"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "675672d04f8ca312-HKG"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "30288064-f78c-4ce6-9715-5e988e405465"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "97"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=WyjL4geLAUKOYj%2FTBvt2ZFZIrxkFZyUUWbkcvYYxAKBTWjtmzmvYVoZJRqbmLpdtVm9yF60ROYclTuinUqYGnD9ds38CakFW7Zk%2B13HYq2DTpj3q2LzKOQbNb1On2HY%2B\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1094,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-27T14:10:05.968Z",
        "time": 647,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 647
        }
      },
      {
        "_id": "56fa76b125e6f1b4f2ad451a73ff8c3d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/crm/v3/owners/"
        },
        "response": {
          "bodySize": 503,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 503,
            "text": "{\"results\":[{\"id\":\"88812444\",\"email\":\"pgilaga@gmail.com\",\"firstName\":\"Paul\",\"lastName\":\"Ilaga\",\"userId\":25387116,\"createdAt\":\"2021-07-07T13:02:40.805Z\",\"updatedAt\":\"2021-07-27T13:17:02.549Z\",\"archived\":false,\"teams\":[{\"id\":\"3785345\",\"name\":\"Test team 1\"}]},{\"id\":\"88814770\",\"email\":\"pgrilaga@gmail.com\",\"firstName\":\"Paul\",\"lastName\":\"Test\",\"userId\":25444566,\"createdAt\":\"2021-07-07T13:06:23.384Z\",\"updatedAt\":\"2021-07-08T04:31:43.259Z\",\"archived\":false,\"teams\":[{\"id\":\"3785345\",\"name\":\"Test team 1\"}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 27 Jul 2021 14:09:58 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "675672d39fd93d48-HKG"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "c7262de1-f65d-4545-b3fa-41dcbb56de84"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "96"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "x-trace",
              "value": "2B73D4F5C572A50EDCA4D5619CFD3EDC5DFF3355CA000000000000000000"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=oAHnrFepRzbBaLzXVuPqTW40sgCA%2F5UFVSur2oKZbpMTLbPRr05L7%2FN49B9qD1vPHinggOIkTkuXwPLDXyJBhfJBpQDHrS8u8dybecZdC%2BmAQ6BN0iForc3ja3RYD36J\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1172,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-27T14:10:06.629Z",
        "time": 897,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 897
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
