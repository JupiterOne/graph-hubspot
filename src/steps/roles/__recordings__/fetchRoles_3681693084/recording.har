{
  "log": {
    "_recordingName": "fetchRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "3befe0ca00fe7532a452821224793de3",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            },
            {
              "name": "accept",
              "value": "application/json"
            }
          ],
          "headersSize": 376,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/settings/v3/users/roles"
        },
        "response": {
          "bodySize": 202,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 202,
            "text": "{\"results\":[{\"id\":\"95124\",\"name\":\"TestRole\",\"requiresBillingWrite\":false},{\"id\":\"95389\",\"name\":\"SuperRole\",\"requiresBillingWrite\":true},{\"id\":\"96735\",\"name\":\"LimitedRole\",\"requiresBillingWrite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 27 Jul 2021 14:09:59 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "202"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "675672d6ee933d94-HKG"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "9775970e-c743-466b-b6f4-061b04a8e813"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "94"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=D9B%2BY63zoytJh4NRFBx%2FVolE3gdbprrNL1LmzFx%2Fd960l%2BT7VJNXiCmk25MbdTer2AZt%2FAioFhutmY9zP7PbMASgSBtshGoM%2F3T2GuusQ7gRSx2c1ZINoRLF4z6V1HmM\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1100,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-27T14:10:07.160Z",
        "time": 506,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 506
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
