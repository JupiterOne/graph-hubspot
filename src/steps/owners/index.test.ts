import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { createIntegrationConfig } from '../../../test/config';
import { IntegrationConfig } from '../../config';
import * as owner from '.';
import * as role from '../roles';
import { Relationships } from '../constants';
import { setupHubspotRecording } from '../../../test/recording';

describe('#fetchOwners', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupHubspotRecording({
      directory: __dirname,
      name: 'fetchOwners',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: createIntegrationConfig(),
    });

    await role.fetchRoles(context);
    await owner.fetchOwners(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const owners = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('User'),
    );

    expect(owners.length).toBeGreaterThan(0);
    expect(owners).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'hubspot_user' },
          id: { type: 'string' },
          name: { type: 'string' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
          email: { type: 'string' },
          userId: { type: 'number' },
          username: { type: 'string' },
          archived: { type: 'boolean' },
          active: { type: 'boolean' },
          createdOn: { type: 'number' },
          updatedOn: { type: 'number' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.USER_ASSIGNED_ROLE._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'ASSIGNED' },
          _type: {
            const: 'hubspot_user_assigned_role',
          },
        },
      },
    });
  });
});
