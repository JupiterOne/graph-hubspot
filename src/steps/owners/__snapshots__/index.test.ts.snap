// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchOwners should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "hubspot_role:95124 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "95124",
            "name": "TestRole",
            "requiresBillingWrite": false,
          },
        },
      ],
      "_type": "hubspot_role",
      "createdOn": undefined,
      "displayName": "TestRole",
      "id": "95124",
      "name": "TestRole",
      "requiresBillingWrite": false,
    },
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "hubspot_role:95389 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "95389",
            "name": "SuperRole",
            "requiresBillingWrite": true,
          },
        },
      ],
      "_type": "hubspot_role",
      "createdOn": undefined,
      "displayName": "SuperRole",
      "id": "95389",
      "name": "SuperRole",
      "requiresBillingWrite": true,
    },
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "hubspot_role:96735 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "96735",
            "name": "LimitedRole",
            "requiresBillingWrite": false,
          },
        },
      ],
      "_type": "hubspot_role",
      "createdOn": undefined,
      "displayName": "LimitedRole",
      "id": "96735",
      "name": "LimitedRole",
      "requiresBillingWrite": false,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "hubspot_user:25387116 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "email": "pgilaga@gmail.com",
            "id": "25387116",
            "primaryTeamId": "3785345",
          },
        },
      ],
      "_type": "hubspot_user",
      "createdOn": undefined,
      "displayName": "pgilaga@gmail.com",
      "email": "pgilaga@gmail.com",
      "id": "25387116",
      "name": "pgilaga@gmail.com",
      "primaryTeamId": "3785345",
      "roleId": undefined,
      "username": "pgilaga@gmail.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "hubspot_user:25444566 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "email": "pgrilaga@gmail.com",
            "id": "25444566",
            "primaryTeamId": "3785345",
            "roleId": "96735",
          },
        },
      ],
      "_type": "hubspot_user",
      "createdOn": undefined,
      "displayName": "pgrilaga@gmail.com",
      "email": "pgrilaga@gmail.com",
      "id": "25444566",
      "name": "pgrilaga@gmail.com",
      "primaryTeamId": "3785345",
      "roleId": "96735",
      "username": "pgrilaga@gmail.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "hubspot_owner:88812444 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "archived": false,
            "createdAt": "Wed Jul 07 2021 21:02:40 GMT+0800 (Philippine Standard Time)",
            "email": "pgilaga@gmail.com",
            "firstName": "Paul",
            "id": "88812444",
            "lastName": "Ilaga",
            "teams": Array [
              PublicTeam {
                "id": "3785345",
                "membership": undefined,
                "name": "Test team 1",
              },
            ],
            "updatedAt": "Tue Jul 27 2021 21:17:02 GMT+0800 (Philippine Standard Time)",
            "userId": 25387116,
          },
        },
      ],
      "_type": "hubspot_owner",
      "active": true,
      "archived": false,
      "createdOn": 1625662960000,
      "displayName": "Paul Ilaga",
      "email": "pgilaga@gmail.com",
      "firstName": "Paul",
      "id": "88812444",
      "lastName": "Ilaga",
      "name": "Paul Ilaga",
      "updatedOn": 1627391822000,
      "userId": 25387116,
      "username": "pgilaga@gmail.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "hubspot_owner:88814770 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "archived": false,
            "createdAt": "Wed Jul 07 2021 21:06:23 GMT+0800 (Philippine Standard Time)",
            "email": "pgrilaga@gmail.com",
            "firstName": "Paul",
            "id": "88814770",
            "lastName": "Test",
            "teams": Array [
              PublicTeam {
                "id": "3785345",
                "membership": undefined,
                "name": "Test team 1",
              },
            ],
            "updatedAt": "Thu Jul 08 2021 12:31:43 GMT+0800 (Philippine Standard Time)",
            "userId": 25444566,
          },
        },
      ],
      "_type": "hubspot_owner",
      "active": true,
      "archived": false,
      "createdOn": 1625663183000,
      "displayName": "Paul Test",
      "email": "pgrilaga@gmail.com",
      "firstName": "Paul",
      "id": "88814770",
      "lastName": "Test",
      "name": "Paul Test",
      "updatedOn": 1625718703000,
      "userId": 25444566,
      "username": "pgrilaga@gmail.com",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "IS",
      "_fromEntityKey": "hubspot_owner:88812444 ",
      "_key": "hubspot_owner:88812444 |is|hubspot_user:25387116 ",
      "_toEntityKey": "hubspot_user:25387116 ",
      "_type": "hubspot_owner_is_user",
      "displayName": "IS",
    },
    Object {
      "_class": "IS",
      "_fromEntityKey": "hubspot_owner:88814770 ",
      "_key": "hubspot_owner:88814770 |is|hubspot_user:25444566 ",
      "_toEntityKey": "hubspot_user:25444566 ",
      "_type": "hubspot_owner_is_user",
      "displayName": "IS",
    },
    Object {
      "_class": "ASSIGNED",
      "_fromEntityKey": "hubspot_user:25444566 ",
      "_key": "hubspot_user:25444566 |assigned|hubspot_role:96735 ",
      "_toEntityKey": "hubspot_role:96735 ",
      "_type": "hubspot_user_assigned_role",
      "displayName": "ASSIGNED",
    },
  ],
  "encounteredTypes": Array [
    "hubspot_role",
    "hubspot_user",
    "hubspot_owner",
    "hubspot_owner_is_user",
    "hubspot_user_assigned_role",
  ],
  "numCollectedEntities": 7,
  "numCollectedRelationships": 3,
}
`;
